app/controllers/posts_controller.rb

```diff
@@ -4,8 +4,30 @@ def index
     @post = Post.all
   end
 
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @post.commentscounter = 0
+    @post.likescounter = 0
+    @post.author = current_user
+
+    if @post.save
+
+      redirect_to "/users/#{current_user.id}/posts", notice: 'Post created successfully!'
+    else
+      render :new, notice: 'Post could not be created.'
+    end
+  end
+
   def show
     @post = Post.find_by_id(params[:id])
     @user = User.find_by_id(params[:user_id])
   end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```

Comment:
- [OPTIONAL] It is a good practice to put your params inside private in order to control the access to your methods. Kindly put that.